ethstorage/es-node
# Move init data file functionality out of run script

The background is that as more features are supported, the configurations become more and more complex, and maintaining the initialization and startup nodes in one script file becomes increasingly unsustainable.

So we separate the init from run so that the flags can be managed separately.

Also supported `--miner.zkey` as path to address https://github.com/ethstorage/es-node/issues/266.

Related tutorials change - https://github.com/ethstorage/ethstorage-doc/pull/32.

Tests need to be covered:
- `init.sh`: data files created and zkey downloaded, snarkjs installation check in all installation options:
  - pre-built
  - from source
  - use pre-built Docker 
  - use local Docker
    - docker-compose
    - run-docker.sh
- `run.sh` starts es-node and mining successfully with `--miner.zkey` as the file path in all the above 3 options.
  - especially in Docker, there is no need to download zkey for a new build as it is mapped to the host disk
